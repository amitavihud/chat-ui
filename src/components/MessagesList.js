import React from 'react'
import styled from "styled-components";
import Message from './Message'
import autoscroll from 'autoscroll-react'

const ScrollableArea = styled.div`
  overflow-y: auto;
`

class MessagesList extends React.Component {
  render () {
    return (
      <ScrollableArea {...this.props}>
        {this.props.messages.map((message, index) => (
          <Message key={index} message={message} me={message.sender === this.props.userId} />
        ))}
      </ScrollableArea>
    )
  }
}

export default autoscroll(MessagesList)