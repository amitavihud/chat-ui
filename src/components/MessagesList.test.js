import React from 'react'
import renderer from 'react-test-renderer';
import MessagesList from './MessagesList'

describe('MessagesList component', () => {
  it('renders the tree', () => {
    const tree = renderer.create(
      <MessagesList messages={[]} />
    )

    expect(tree).toMatchSnapshot()
  })

  describe('there are some messages', () => {
    it('renders them', () => {
      const tree = renderer.create(
        <MessagesList messages={[
          { text: '1', username: 'Donald', avatar: 'someurl.com/1.png' },
          { text: '1', username: 'Kim', avatar: 'someurl.com/2.png' }
        ]} />
      )

      expect(tree).toMatchSnapshot()
    })
  })
})