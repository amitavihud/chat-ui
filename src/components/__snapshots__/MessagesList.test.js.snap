// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessagesList component renders the tree 1`] = `
<div
  className="sc-EHOje faCQco"
  onScroll={[Function]}
/>
`;

exports[`MessagesList component there are some messages renders them 1`] = `
<div
  className="sc-EHOje faCQco"
  onScroll={[Function]}
>
  <div
    className="sc-bdVaJa eMAKFS"
  >
    <img
      src="someurl.com/1.png"
    />
    <div
      className="sc-bwzfXH gheIUh"
    >
      <div
        className="sc-htpNat jQRWOW"
        color="#F2F2FF"
      >
        <div
          className="sc-bxivhb CqVgY"
        />
        <span
          className="sc-ifAKCX BnkYH"
        >
          Donald
          :
        </span>
        1
      </div>
    </div>
  </div>
  <div
    className="sc-bdVaJa eMAKFS"
  >
    <img
      src="someurl.com/2.png"
    />
    <div
      className="sc-bwzfXH gheIUh"
    >
      <div
        className="sc-htpNat jQRWOW"
        color="#F2F2FF"
      >
        <div
          className="sc-bxivhb CqVgY"
        />
        <span
          className="sc-ifAKCX BnkYH"
        >
          Kim
          :
        </span>
        1
      </div>
    </div>
  </div>
</div>
`;
